File: common/retex/src/main/kotlin/com/himamis/retex/renderer/android/resources/ResourceLoaderFactoryAndroid.java
package com.himamis.retex.renderer.android.resources;

import android.content.res.AssetManager;

import com.himamis.retex.renderer.share.platform.resources.ResourceLoader;
import com.himamis.retex.renderer.share.platform.resources.ResourceLoaderFactory;

public class ResourceLoaderFactoryAndroid implements ResourceLoaderFactory {

    private AssetManager mAssetManager;

    public ResourceLoaderFactoryAndroid(AssetManager assetManager) {
        mAssetManager = assetManager;
    }

    public ResourceLoader createResourceLoader() {
        return new ResourceLoaderA(mAssetManager);
    }

}

File: modules/retex/src/androidMain/kotlin/com/himamis/retex/renderer/android/resources/ResourceLoaderFactoryAndroid.kt
package com.himamis.retex.renderer.android.resources

import android.content.res.AssetManager
import com.himamis.retex.renderer.share.platform.resources.ResourceLoader
import com.himamis.retex.renderer.share.platform.resources.ResourceLoaderFactory

class ResourceLoaderFactoryAndroid(private val mAssetManager: AssetManager) :
    ResourceLoaderFactory {
    override fun createResourceLoader(): ResourceLoader {
        return ResourceLoaderA(mAssetManager)
    }
}
